.github/workflows/coverage-check.yml

```diff
@@ -0,0 +1,100 @@
+name: Check Test Coverage
+
+on:
+  pull_request:
+    branches: [development]
+
+jobs:
+  test-coverage:
+    name: Check Test Coverage
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout development branch
+        uses: actions/checkout@v3
+        with:
+          ref: development
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+
+      - name: Install dependencies
+        run: sudo apt -y update && sudo apt -y install libusb-1.0-0-dev libudev-dev
+
+      - name: Yarn Install
+        run: yarn install --mode=skip-build && yarn allow-scripts
+
+      - name: Setup Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Setup Forge
+        run: yarn workspace @ubiquity/contracts forge:install
+
+      - name: Get development branch coverage
+        run: |
+          # this is temporary, until we merge the pr branch into development then we can use the yarn workspaces command
+
+          cd packages/contracts
+
+          COVERAGE_DEVELOPMENT_OUTPUT=$(forge coverage)
```

Comment:
so this only works for local testing all the remote tests are ignored also it includes the test helpers in the coverage 
IMHO would be better to have a runner script like runner.st to ensure that only the require files are coverage tested

Reply:
ok, so I have a question if `forge coverage` only works for local tests could u suggest another which works for remote testing I couldn't find one

Reply:
IMO the most important is to remove the unnecessary files from the coverage 
then ideally all the remote testing should be here to test some scenarios (integration testing) but don't remove the need for unit testing .
I am not saying that forge coverage only works locally I am saying that as it is in this action only local testing are included in the test coverage. btw forge coverage does include a `--fork-url` option e.g `forge coverage --rpc-url http://localhost:8545` so we should be able to add these tests to the coverage action

Reply:
ok I think I get what you mean, I will prefer to use a step though to remove the files in utils

Reply:
Hi so there is this option on forge coverage `--no-match-path` so I can do this to exclude file under the utils directory 
`--no-match-path=**/utils/**=**/utils/**`. Also for the `--rpc-url` option I just wanted to know as part of steps for that I will need to setup anvil right?

Reply:
You can check on the development branch there is already an action forking the mainnet to run tests

Reply:
u are talking of setup-anvil.yml

Reply:
Hi @zgorizzo69 can you check the PR again